drv/lpc55-update-server/src/main.rs

```diff
@@ -279,31 +277,133 @@ fn validate_header_block(
     Ok(())
 }
 
+/// Performs an erase-write sequence to a single page within a given target
+/// image.
 fn do_block_write(
     img: UpdateTarget,
     block_num: usize,
     flash_page: &mut [u8; BLOCK_SIZE_BYTES],
 ) -> Result<(), UpdateError> {
-    let result = unsafe {
-        // The write_to_flash API takes raw pointers due to TrustZone
-        // ABI requirements which makes this function unsafe.
-        tz_table!().write_to_flash(
-            img,
-            block_num as u32,
-            flash_page.as_mut_ptr(),
-        )
+    let page_num = block_num as u32; // TODO why not use the same type
```

Comment:
I'm totally fine changing the type in the function parameter, although it would ripple upward at least a bit.

Reply:
Ah, yeah, I meant to propagate this before pushing the PR -- the TODO was for me. Lemme take a pass at this and see how it looks.

Reply:
So it turns out the IDL interface uses `usize` here, and changing this would be more work. Will instead remove the TODO.